runtime/libpgmath/lib/common/fltmanip.c

```diff
@@ -50,6 +71,130 @@ __remainderf(float x, float y)
   return __mth_i_remainder(x, y);
 }
 
+#ifdef TARGET_SUPPORTS_QUADFP
+static inline int
+isnan(unsigned int h, unsigned int j, unsigned int k, unsigned int l)
+{
+    return (((h & 0x7fff0000) == 0x7fff0000) &&
+           ((l | k | j | (h & 0xffff)) != 0));
+}
+#endif
+
+#define _MAXLONGDOUBLE (1.18973149535723176508575932662800702E+4932L)
+#define IX_SIZE 4 /* ix[] has 4 element and 128 bits */
+#define IZ_SIZE 2 /* iz[] has 4 element and 128 bits */
+#define GET_SIGN 31
+#define GET_EXP1 16
+#define GET_EXP2 23
+#define IS_ZERO(h, j, k, l) (((h) & 0x7fffffff) == 0) && (((l) | (k) | (j)) == 0)
+#define IS_MIN_NORM(h, j, k, l, s) (((h) & 0x7fffffff) == 0x00010000) && (((l) | (k) | (j)) == 0) && (s)
+#define IS_MAX_NORM(h, j, k, l ,s) (((h) & 0x7fffffff) == 0x7ffeffff) && (((~((l) & 0xffffffff)) | \
+                           (~((k) & 0xffffffff)) | (~((j) & 0xffffffff))) == 0) && (!(s))
+#ifdef TARGET_SUPPORTS_QUADFP
+long double
+__nextafterl(long double x, long double y)
+{
+  long double ex;
+  unsigned int ix[IX_SIZE], ixh, ixj, ixk, ixl, iyh, iyj, iyk, iyl;
+  unsigned long long iz[IZ_SIZE], izh, izl;
+  int idflag, subflag;
+
+  ex = y;
+  iyl = *((int*)(&(ex)));
```

Comment:
Thanks, because the code already commited to master branch, I filed a new PR https://github.com/flang-compiler/flang/pull/1132 to fix the format issue.